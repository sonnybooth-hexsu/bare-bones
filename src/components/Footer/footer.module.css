@value colors: "../../styles/variables.module.css";
@value mqs: "../../styles/mqs.module.css";
@value colorPrimary, colorBlack, colorWhite from colors;
@value fromMedium from mqs;

.footerContainer {
    background-color: colorBlack;
}

.footer {
    display: flex;
    flex-direction: column;
    text-align: center;
    padding: 4rem 0;
    color: colorWhite;
    
    @media (fromMedium) {
        flex-direction: row;
    }
}

.footerLogoContainer {
    @media (fromMedium) {
        text-align: left;
        align-self: flex-start;
        flex-grow: 1;
    }
}

.footerLogo {
    display: inline-block;
    align-self: center;
    background-color: colorWhite;
    color: colorBlack;
    padding: 1rem;
    font-size: 1.4rem;
    margin-bottom: 1rem;
}

.footerNav {
    margin-top: 3rem;

    @media (fromMedium) {
        margin-top: 0;
        margin-right: 6rem;
    }
}

.footerNav li {
    margin-bottom: 0.5rem;
}

.footerNav a:hover {
    text-decoration: underline;
}

.footerMeta {
    margin-top: 3rem;

    @media (fromMedium) {
        margin-top: 0;
        font-size: 0.75rem;
    }
}