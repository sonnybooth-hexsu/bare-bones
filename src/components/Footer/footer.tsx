import React from "react"
import styles from "./footer.module.css"

type FooterProps = {
  siteTitle: string
  navLinks: Array<{ name: string; page: string; id: number }>
  description: string
  email: string
  telephone: string
}

export const Footer = ({
  siteTitle,
  navLinks,
  description,
  email,
  telephone,
}: FooterProps) => {
  return (
    <div className={styles.footerContainer}>
      <div className="container mx-auto pl-6 pr-6 md:pl-0 md:pr-0">
        <div className="pl-4 pr-4">
          <div className={styles.footer}>
            <div className={styles.footerLogoContainer}>
              <a href="/" className={styles.footerLogo}>
                {siteTitle}
              </a>
              <p data-testid="footerNavigationDescription">{description}</p>
            </div>
            <div>
              <nav className={styles.footerNav}>
                <ol>
                  {navLinks.map(({ name, page }, i) => (
                    <li data-testid="footerNavigationLink" key={i}>
                      <a href={page}>{name}</a>
                    </li>
                  ))}
                </ol>
              </nav>
            </div>
            <div className={styles.footerMeta}>
              <p>{email}</p>
              <p>{telephone}</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Footer
