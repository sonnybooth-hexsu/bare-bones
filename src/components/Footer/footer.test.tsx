/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import { queryAllByTestId, getByText, getByTestId } from "@testing-library/dom"
import Footer from "./footer"

const testNavLinks = [
  {
    name: `Link 1`,
    page: `link-1`,
    id: 1,
  },
  {
    name: `Link 2`,
    page: `link-2`,
    id: 2,
  },
  {
    name: `Link 3`,
    page: `link-3`,
    id: 3,
  },
]

afterEach(() => {
  cleanup()
})

describe("<Footer />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <Footer
        siteTitle="Test Site"
        navLinks={testNavLinks}
        description="I am test description"
        email="test@email.com"
        telephone="+441111111111"
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should have the length of links passed through", () => {
    const { container } = render(
      <Footer
        siteTitle="Test Site"
        navLinks={testNavLinks}
        description="I am test description"
        email="test@email.com"
        telephone="+441111111111"
      />
    )

    expect(queryAllByTestId(container, "footerNavigationLink")).toHaveLength(
      testNavLinks.length
    )
  })

  it("should have the description text passed through", () => {
    const { container } = render(
      <Footer
        siteTitle="Test Site"
        navLinks={testNavLinks}
        description="I am test description"
        email="test@email.com"
        telephone="+441111111111"
      />
    )

    const descriptionEl = getByTestId(container, "footerNavigationDescription")

    expect(getByText(descriptionEl, "I am test description"))
  })
})
