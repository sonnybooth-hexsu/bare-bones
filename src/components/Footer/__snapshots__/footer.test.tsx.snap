// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<div>
  <div>
    <div
      class="bg-black"
    >
      <div
        class="container mx-auto pl-6 pr-6 pt-16 pb-16"
      >
        <div
          class="pl-4 pr-4"
        >
          <div
            class="grid grid-cols-1 gap-10 md:gap-6 md:grid-cols-3 lg:grid-cols-12"
          >
            <div
              class="flex items-center justify-center text-white md:col-span-3 lg:col-span-4 lg:items-left"
            >
              <div
                class="text-center lg:text-left"
              >
                <a
                  class="bg-white text-black pt-4 pb-4 pr-6 pl-6"
                  href="/"
                >
                  Test Site
                </a>
                <p
                  class="mt-4 mb-10 text-sm"
                  data-testid="footerNavigationDescription"
                >
                  I am test description
                </p>
                <div>
                  <svg
                    class="inline mr-2 ml-2 text-white"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M18 2h-3a5 5 0 0 0-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 0 1 1-1h3z"
                    />
                  </svg>
                  <svg
                    class="inline mr-2 ml-2 text-white"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
                    />
                  </svg>
                  <svg
                    class="inline mr-2 ml-2 text-white"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
                    />
                    <rect
                      height="12"
                      width="4"
                      x="2"
                      y="9"
                    />
                    <circle
                      cx="4"
                      cy="4"
                      r="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="lg:col-span-2 lg:col-start-6"
            >
              <nav
                class="text-center lg:text-left"
              >
                <h6
                  class="text-sm text-white lg:text-left"
                >
                  Title
                </h6>
                <ol>
                  <li
                    data-testid="footerNavigationLink"
                  >
                    <a
                      class="text-sm text-center text-white"
                      href="link-1"
                    >
                      Link 1
                    </a>
                  </li>
                  <li
                    data-testid="footerNavigationLink"
                  >
                    <a
                      class="text-sm text-center text-white"
                      href="link-2"
                    >
                      Link 2
                    </a>
                  </li>
                  <li
                    data-testid="footerNavigationLink"
                  >
                    <a
                      class="text-sm text-center text-white"
                      href="link-3"
                    >
                      Link 3
                    </a>
                  </li>
                </ol>
              </nav>
            </div>
            <div
              class="lg:col-span-2"
            >
              <nav
                class="text-center lg:text-left"
              >
                <h6
                  class="text-sm text-white lg:text-left"
                >
                  Title
                </h6>
                <ol>
                  <li>
                    <a
                      class="text-sm text-center text-white "
                      href="link-1"
                    >
                      Link 1
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-sm text-center text-white "
                      href="link-2"
                    >
                      Link 2
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-sm text-center text-white "
                      href="link-3"
                    >
                      Link 3
                    </a>
                  </li>
                </ol>
              </nav>
            </div>
            <div
              class="lg:col-span-2"
            >
              <nav
                class="text-center"
              >
                <h6
                  class="text-sm text-white lg:text-left"
                >
                  Get in touch
                </h6>
                <div>
                  <p
                    class="text-sm text-center text-white lg:text-left"
                  >
                    test@email.com
                  </p>
                  <p
                    class="text-sm text-center text-white lg:text-left"
                  >
                    +441111111111
                  </p>
                </div>
              </nav>
            </div>
          </div>
        </div>
      </div>
      <div
        class="bg-gray-700"
      >
        <div
          class="container mx-auto pl-6 pr-6"
        >
          <div
            class="pl-4 pr-4"
          >
            <p
              class="text-xs pt-2 pb-2 text-center text-white lg:text-left"
            >
              Â©
              2020
               - Bare Bones | All right reserved.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
